### ISSUE TITLE
Addition of a build system generator

### ISSUE THREAD

**Issue by elfring** [e-n-f/datamaps#8]
I suggest to reuse a higher level build system than your current [small make file](https://github.com/ericfischer/datamaps/blob/09c6b67912d4840827bd8b01e6384d6942758f2e/Makefile#L1) so that powerful checks for software features will become easier.
- [CMake](http://cmake.org/)
- [Autotools](http://www.gnu.org/software/autoconf/#TOCintroduction)

**Comment by e-n-f** [comment]
The Makefile is already more complicated than I want it to be, actually. I'd like to get rid of the pkg-config stuff and just say -lpng if I could get away with it.

**Comment by elfring** [comment]
Are you interested in portable make files?
How do you think about to improve software portability by feature checks?

**Comment by e-n-f** [comment]
No, I would in general like to depend only on facilities that exist on any reasonably modern Unix system.

**Comment by elfring** [comment]
There are some variations to consider for implementation details to be safe ...

**Comment by e-n-f** [comment]
As far as I know, everything I am doing is platform-neutral for the platforms I am targeting.

**Comment by elfring** [comment]
Does [a tool like "autoscan"](https://www.gnu.org/savannah-checkouts/gnu/autoconf/manual/autoconf-2.69/html_node/autoscan-Invocation.html) show any software dependencies that you find worth for further considerations?

**Comment by noname2534** [comment]
+1
